### DISCUSSION TITLE
大佬们请问生产者组（producerGroup）是干嘛的？

### DISCUSSION THREAD

**Discussion by liuweiGL** [apache/rocketmq#7387] [Q&A] [answered]
我看 spring cloud alibaba 的示例时发现有个 group 配置 
https://github.com/alibaba/spring-cloud-alibaba/blob/2823ae7443f05faa075a94cf5375919b627ee08d/spring-cloud-alibaba-examples/rocketmq-example/rocketmq-broadcast-example/rocketmq-broadcast-producer-example/src/main/resources/application.yml#L14
在 `DefaultMQProducer` 中也发现了 `producerGroup` 属性，注释说是事务用到的。但是我看[官网](https://rocketmq.apache.org/zh/docs/domainModel/04producer)只有消费者组的解释，没有提到生产者组概念。请教一下这个我到底要不要配置呢，不配置会对事务有什么影响啊？

**Comment by RongtongJin** [comment, marked as answer]
主要是用于事务消息回查的。事务消息的回查机制，回查时 Broker 端若发现原始生产者连接已经断开，则会联系同一生产者组（producerGroup）的其他生产者实例以 Commit 或 Rollback 半事务消息。
